test/higher-order.jsx

```diff
@@ -195,6 +195,40 @@ describe('Higher Order', function() {
       }, 50);
     });
 
+    it('should not re-render the component if cursor value hasnt changed.', function(done) {
+      const tree = new Baobab({name: 'Sherlock', address: {apt: "221B", street: "Baker St"}}, {asynchronous: false});
+
+      var numRendered = 0
+
+      class Child extends Component {
+        render() {
+          numRendered++
+
+          return (
+            <span>
+              {this.props.name}, stays at {this.props.address.apt} {this.props.address.street}
+            </span>
+          );
+        }
+      }
+
+      const Root = root(tree, BasicRoot);
+
+      const BranchedChild = branch({
+        name: 'name',
+        address: 'address'
+      }, Child);
+
+      const wrapper = mount(<Root tree={tree}><BranchedChild /></Root>);
+
+      tree.set('address', {apt: "221B", street: "Baker St"});
```

Comment:
I can understand why baobab wouldn't catch this case by default.
I'm curious about if you had done:
```
tree.set(['address', 'apt'], '221B)
```
whether that would have caused an extra render or not.
Can you add a test for that / test that case without the new shouldComponentUpdate?